### PR TITLE
K8SPSMDB-496: fix custom config hash annotations

### PR DESCRIPTION
**Pull request by ollevche**
[K8SPSMDB-496](https://jira.percona.com/browse/K8SPSMDB-496) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPSMDB-496

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| smart-update | passed |
| version-service | passed |
| operator-self-healing-chaos | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |
| pitr | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/732/commits/cd2cda8853b8e28eeac194ad071adb232a7a22ea
image: `perconalab/percona-server-mongodb-operator:PR-732-cd2cda88`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/732/commits/cbffe1f6536914f8a9a6e31e14a8371e87e22aa5
image: `perconalab/percona-server-mongodb-operator:PR-732-cbffe1f6`

File: pkg/psmdb/mongos.go [hunk 1]
```diff
@@ -64,8 +63,8 @@ func MongosDeploymentSpec(cr *api.PerconaServerMongoDB, operatorPod corev1.Pod,
 		annotations = make(map[string]string)
 	}
 
-	if c := cr.Spec.Sharding.Mongos.Configuration; c != "" && configSource == VolumeSourceConfigMap {
-		annotations["percona.com/configuration-hash"] = fmt.Sprintf("%x", md5.Sum([]byte(c)))
+	if customConf.Type.IsUsable() {
```
Comment (qJkee) [inline pkg/psmdb/mongos.go:66]: but this config provided by user directly in cr. We should handle it too

File: pkg/psmdb/mongos.go [hunk 2]
```diff
@@ -64,8 +63,8 @@ func MongosDeploymentSpec(cr *api.PerconaServerMongoDB, operatorPod corev1.Pod,
 		annotations = make(map[string]string)
 	}
 
-	if c := cr.Spec.Sharding.Mongos.Configuration; c != "" && configSource == VolumeSourceConfigMap {
```
Comment (qJkee) [inline pkg/psmdb/mongos.go:67]: but this config provided by user directly in cr. We should handle it too

Comment (ollevche) [reply pkg/psmdb/mongos.go:67]: yes, it's handled from a config map to simplify the logic

configuration from cr.yaml goes to config map and from there it's handled as any other custom configuration

**Review by qJkee — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**